db/db.go

```diff
@@ -1013,10 +1013,12 @@ func saveValidators(data *types.EpochData, tx *sqlx.Tx, client rpc.Client) error
 		return err
 	}
 
-	thresholdSlot := latestBlock - 64
-	if latestBlock < 64 {
-		thresholdSlot = 0
+	var lastFinalizedEpoch uint64
```

Comment:
A query here for the latest finalized epoch should not be necessary. We have services that store frequently used values such has the latest finalized epoch in memory. Check the services.go file.

Reply:
This is correct but not straight forward. Using services in this file causes a circular dependency. I tried that but did not want to touch the architecture for this issue, hence the query.
I guess this is also the reason for the `latestBlock` being queried above to get `latestEpoch` instead of using `services.LatestEpoch`.

How do you want me to proceed?

Reply:
Regarding the use of the `WriterDb` here:
Based on this [answer](https://stackoverflow.com/c/bitfly/a/124/17), the `WriterDb` should be used for read access if the data may have been changed very recently. Within `func saveEpoch`, the function we are looking at (`func saveValidators`) is called and afterwards the epoch table is updated. So yes, the `ReaderDb`should work here. But if this order is changed for whatever reason in the future, using the `ReaderDb` would break this handling and it would be hard to spot that.

I guess there is a similar reason why the `WriterDb` is also used above to access the `latestBlock`.

If you still want me to change that, I will do that of course. To be consistent, I would suggest to use the `ReaderDb`for the `latestBlock` too in that case. I just wanted to share my train of thoughts.

Reply:
Ah I see its part of the exporter and not of the frontend. Then its fine how it is. I would however similar to above remove the threshold epoch and use the lastFinalizedEpoch.

Reply:
Hard to see what functions some of the code belongs to without looking at the whole file..